### ISSUE TITLE
bug: Ionic React with redirect on tab with nested route no longer works

### ISSUE THREAD

**Issue by adamduren** [ionic-team/ionic#20349]
I have a demo repo [here](https://github.com/adamduren/ionic-react-routing-redirect-issue) with full description and video. Simple commit history. Code diff for changes made to tabs starter [here](https://github.com/adamduren/ionic-react-routing-redirect-issue/commit/649c5c95da619bbbfb8c88a46dec2a05bfa0830e)

# Bug Report

**Ionic version:**

[x] **4.11.6** (introduced here)
[x] **4.11.7**
[x] **4.11.8**
[x] **4.11.9**
[x] **4.11.10** (regressed further to where page does not even display)

**Demo:**
![](https://github.com/adamduren/ionic-react-routing-redirect-issue/raw/master/tab-issue.gif)

**Current behavior:**
  1. Correct: TAB_WITH_REDIRECT is displayed after steps to reproduce.
  1. Incorrect: TAB_WITHOUT_REDIRECT is displayed after steps to reproduce.

**Expected behavior:** (Multiple Scenarios)
  1. TAB_WITH_REDIRECT is displayed after steps to reproduce.
  1. TAB_WITH_REDIRECT is displayed after steps to reproduce.

**Steps to reproduce:**
 1. Navigate via address bar to the root url of TAB_WITH_REDIRECT. By clicking, switch to a TAB_WITHOUT_REDIRECT. By clicking, switch back to TAB_WITH_REDIRECT. Observe TAB_WITH_REDIRECT is displayed.
  1. Navigate via address bar to the url of a TAB_WITHOUT_REDIRECT. Switch by clicking, to the tab TAB_WITH_REDIRECT. Switch by clicking, back to TAB_WITHOUT_REDIRECT. Switch by clicking, back to TAB_WITH_REDIRECT. Observe TAB_WITH_REDIRECT is displayed.

**Related code:**
See [repo](https://github.com/adamduren/ionic-react-routing-redirect-issue)

**Ionic info:**

```
Ionic:

   Ionic CLI       : 5.4.15 (/Users/adamduren/.nvm/versions/node/v10.14.1/lib/node_modules/ionic)
   Ionic Framework : @ionic/react 4.11.10

Utility:

   cordova-res (update available: 0.9.0) : 0.6.0
   native-run (update available: 0.3.0)  : 0.2.8

System:

   NodeJS : v10.14.1 (/Users/adamduren/.nvm/versions/node/v10.14.1/bin/node)
   npm    : 6.13.2
   OS     : macOS Catalina
```

**Comment by OoDeLally** [comment]
I probably have the same problem. It appears in `"@ionic/react-router": "4.11.8"`.

https://github.com/OoDeLally/ion-router-redirect-bug

**Comment by adamduren** [comment]
@brandyscarney what's the usual time for triaging bugs? This is a pretty big showstopper for us. Luckily we found a working patch version.

**Comment by elylucas** [comment]
Hi @adamduren,

I've looked through the repo you provided (thanks for that!), and theres definitely some things in there that aren't quite supported. For instance, all the routes for the tabs should be defined in the same IonRouterOutlet, where I see some nested routes in some of the sub components.

However, I'm having a hard time trying to discern what is going on though. Could you provide a use case (just in words, no code) about what it is you are tryng to do?

Thanks

**Comment by adamduren** [comment]
My first objective is I wanted to break the navigation of each tab into its own file.
Secondly I wanted to have a tab called admin. Which would have the URL `tabs/admin`. This tab would default to `tabs/admin/initiatives` if the user tried to go to `tabs/admin` directly versus having the Initiatives admin page location at `tabs/admin`.

**Comment by elylucas** [comment]
Hi @adamduren,

We had some fixes in 5.0.1 that might of fixed this issue. Could you test again and let me know if it still happens?

Thanks

**Comment by adamduren** [comment]
@elylucas just upgraded to v5 and it appears to work as expected. Thank You!
